Assets/Scripts/FogOfWar.cs

```diff
@@ -0,0 +1,138 @@
+﻿using UnityEngine;
+using UnityEditor;
+using System.Collections;
+using System.Collections.Generic;
+
+public class FogOfWar : MonoBehaviour
+{
+	public GameObject[] cloudPrefabs;
+	public GameObject colliderPrefab;
+
+	public float radius;
+	public float radiusError;
+	public float height;
+
+	[Range( 0.0f, 360.0f )]
+	public float cloudIncrement;
+
+	public float minScale;
+	public float maxScale;
+
+	public int numClouds;
+	public int numColliders;
+
+	private new Transform transform;
+
+	public void Awake()
+	{
+		transform = GetComponent<Transform>();
+	}
+
+	void Start()
+	{
+		GenerateFogOfWar();
+	}
+
+	public void GenerateFogOfWar()
+	{
+		ClearFogOfWar();
+		GenerateClouds();
+		GenerateColliders();
+	}
+
+	public void ClearFogOfWar( bool inEditor = false )
+	{
+		// destroy all children
+		var children = new List<GameObject>();
+		foreach ( Transform child in transform ) children.Add( child.gameObject );
```

Comment:
on the `foreach` loop? That's something that I copied from an example. I didn't bother splitting it into multiple lines but if you don't want to do one-line ifs and loops then I'll break it up.